src/conn/conn_log.c

```diff
@@ -400,12 +394,17 @@ __log_remove_once(WT_SESSION_IMPL *session, uint32_t backup_file)
     min_lognum = backup_file == 0 ? WT_MIN(log->ckpt_lsn.l.file, log->sync_lsn.l.file) :
                                     WT_MIN(log->ckpt_lsn.l.file, backup_file);
 
+    __wt_readlock(session, &conn->debug_log_retention_lock);
+
     /* Adjust the number of log files to retain based on debugging options. */
-    WT_ORDERED_READ(dbg_val, conn->debug_ckpt_cnt);
-    if (FLD_ISSET(conn->debug_flags, WT_CONN_DEBUG_CKPT_RETAIN) && dbg_val != 0)
-        min_lognum = WT_MIN(conn->debug_ckpt[dbg_val - 1].l.file, min_lognum);
-    WT_ORDERED_READ(dbg_val, conn->debug_log_cnt);
-    if (dbg_val != 0) {
+
+    dbg_ckpt_cnt = conn->debug_ckpt_cnt;
+    dbg_ckpt_retain = FLD_ISSET(conn->debug_flags, WT_CONN_DEBUG_CKPT_RETAIN);
+    if (dbg_ckpt_retain && dbg_ckpt_cnt != 0)
```

Comment:
We could also remove `debug_ckpt_retain`.

Reply:
Updated in [ad69de5](https://github.com/wiredtiger/wiredtiger/pull/9428/commits/ad69de5c0b8587bece045368322a982d3de4c89a)